reporting/server/session.h

```diff
@@ -251,10 +311,14 @@ class session : public std::enable_shared_from_this<session>
             BOOST_LOG_TRIVIAL(debug) << " received request : " << request;
             try
             {
+                if (subscriptions_.expired())
+                    return;
+
                 response = buildResponse(
                     request, 
                     backend_,
-                    subscriptions_,
+                    subscriptions_.lock(),
+                    balancer_,
```

Comment:
The proper way to do this is to call `subscriptions_.lock()`, check the result to make sure it's not `nullptr`, and then pass that in to `buildResponse`. The way it is now, `expired()` could return false, then all of the shared_ptr instances get deleted, then we call `lock`, which is going to return `nullptr`.